conference.js

```diff
@@ -139,14 +139,22 @@ const eventEmitter = new EventEmitter();
 let room;
 let connection;
 
+// Used to be able to initialize early and wait for _connectionPromise without relying on async code to create
```

Comment:
Maybe use a Deferred? We have an implementation in LJM.

Reply:
yeah. The only problem would be await doesn't work with that I guess, but it's not hard to change. Probably a better place for Deferred would be in the js utils lib.

Anyway, this fix may not work fully for you... I was testing it only with iframe case where we recreate it on connect failed attempt. I dont know what will happen if connection fails (not even sure if it's handled on prejoin correctly). Someone from the team would have to take a closer look at it anyway. Please treat this PR more as a bug report rather than a proper fix.

Reply:
Ack. FWIW the Deferred I added can be awaited so we could use that if need be.